src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -570,6 +569,24 @@ CreateShardsOnWorkersViaExecutor(Oid distributedRelationId, List *shardPlacement
 		taskList = lappend(taskList, task);
 	}
 
+	if (useExclusiveConnection)
+	{
+		/*
+		 * When the table has local data, we force max parallelization so data
+		 * copy is done efficiently.
+		 */
+		SetLocalForceMaxQueryParallelization();
```

Comment:
We could get here even if we're not inside a transaction block. And, actually `SET LOCAL` gives a WARNING executed outside of a transaction block (`WARNING:  SET LOCAL can only be used in transaction blocks`). However, the warning is given from one level higher function of `set_config_option `.

Still, reading the postgres code, I think there is no harm to call `set_config_option ` even if we're not inside a transaction block. It also does what we want.

@marcocitus any opinions on this? Note that with `SetLocalMultiShardModifyMode()` we're always in a transaction block.

Reply:
I think it should be ok.